src/arkode/arkode.c

```diff
@@ -2972,14 +3006,17 @@ int arkCheckConstraints(ARKodeMem ark_mem, int *constrfails, int *nflag)
   /* Return with error if using fixed step sizes */
   if (ark_mem->fixedstep) return(ARK_CONSTR_FAIL);
 
-  /* Return with error if |h| == hmin */
-  if (SUNRabs(ark_mem->h) <= ark_mem->hmin*ONEPSM) return(ARK_CONSTR_FAIL);
-
-  /* Reduce h by computing eta = h'/h */
+  /* Recommend step size adjustment to satisfy constraints */
   N_VLinearSum(ONE, ark_mem->yn, -ONE, ark_mem->ycur, tmp);
   N_VProd(mm, tmp, tmp);
-  ark_mem->eta = RCONST(0.9)*N_VMinQuotient(ark_mem->yn, tmp);
-  ark_mem->eta = SUNMAX(ark_mem->eta, TENTH);
+  hnew = ark_mem->h * SUNMAX(RCONST(0.9)*N_VMinQuotient(ark_mem->yn, tmp), TENTH);
```

Comment:
Could this heuristic be handled by the SUNHeuristic object?

Reply:
I don't think so, since this calculation inherently relies on N_Vectors that encode the degree of constraint violation.  Perhaps the hard-coded value of `0.9` could be parameterized, but I think that the rest of this calculation should reside here inside `arkCheckConstraints`.

@gardner48, do you have any thoughts on this?

Reply:
I think we should parameterize the `0.9`. We could create a controller type (see [this comment](https://github.com/LLNL/sundials/pull/289#discussion_r1359967838)) for determining the new step size with constraints, possibly doing something different from the current linear interpolant, but that could be done in a later PR.